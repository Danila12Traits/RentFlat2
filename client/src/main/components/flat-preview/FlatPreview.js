import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Slider from '~/common/components/slider/Slider';
import './style.less';

class FlatPreview extends Component {
  static propTypes = {
    flat: PropTypes.shape({
      photoUrls: PropTypes.arrayOf(PropTypes.string),
      price: PropTypes.number,
      _id: PropTypes.string,
    }),
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }).isRequired,
  };

  static defaultProps = {
    flat: {
      photoUrls: [],
      price: 0,
      _id: '',
    },
  };

  constructor(props) {
    super(props);

    this.state = {
      photoUrlIndex: 0,
    };
  }

  get img() {
    const { photoUrls } = this.props.flat;
    const { photoUrlIndex } = this.state;

    return (<img
      className="FlatPreview__img"
      alt="flat"
      src={photoUrls[photoUrlIndex]}
    />);
  }


  setPhotoUrlIndex = (index = 0) => {
    this.setState({
      photoUrlIndex: index,
    });
  }

  redirectToFlatDetails = () => {
    const { _id } = this.props.flat;
    this.props.history.push(`/flat-details/${_id}`);
  }

  render() {
    const { price, photoUrls } = this.props.flat;
    const { photoUrlIndex } = this.state;

    return (
      <div className="FlatPreview">
        <span onClick={this.redirectToFlatDetails}>{this.img}</span>
        <div>Text</div>
        <div>Text2</div>
        <div className="FlatPreview__priceLabel">$ {price}</div>
        <Slider
          photoUrls={photoUrls}
          setPhotoUrlIndex={this.setPhotoUrlIndex}
          photoUrlIndex={photoUrlIndex}
        />
      </div>
    );
  }
}

export default FlatPreview;
