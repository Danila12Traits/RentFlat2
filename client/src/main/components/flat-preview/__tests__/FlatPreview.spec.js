import React from 'react';
import renderer from 'react-test-renderer';
import FlatPreview from '../FlatPreview';

const flat = {
  _id: '57a459a18e126011006e10d7',
  photoUrls: [
    'https://flatbel.s3.amazonaws.com/1470388637305_20160607_180802-1024x1820.jpg',
  ],
  price: 25,
  address: {
    latitude: '53.8622209',
    longitude: '27.46885659999998',
  },
};

const history = {
  push: () => {},
};

describe('FlatPreview component', () => {
  it('renders default layout', () => {
    const renderedComponent = renderer.create(<FlatPreview flat={flat} history={history} />);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
    renderedComponent.unmount();
  });
});
