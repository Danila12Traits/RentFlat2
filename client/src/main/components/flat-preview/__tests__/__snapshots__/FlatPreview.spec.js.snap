// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatPreview component renders default layout 1`] = `
<div
  className="FlatPreview"
>
  <span
    onClick={[Function]}
  >
    <img
      alt="flat"
      className="FlatPreview__img"
      src="https://flatbel.s3.amazonaws.com/1470388637305_20160607_180802-1024x1820.jpg"
    />
  </span>
  <div>
    Text
  </div>
  <div>
    Text2
  </div>
  <div
    className="FlatPreview__priceLabel"
  >
    $ 
    25
  </div>
  <div
    className="Slider"
  >
    <div
      className="Slider__item Slider__item--active"
      onClick={[Function]}
    />
  </div>
</div>
`;
